packages/mobile/src/qrcode/utils.ts

```diff
@@ -95,7 +97,20 @@ export function* handleBarcode(
         displayId: data.e164PhoneNumber,
       }
 
-  yield put(storeLatestInRecents(recipient))
+  // TODO data in the QR should be encrypted so noone can create an arbitrary QR with any phone and address combination
+  // add phone number to the recipientCache
+  recipientCache[data.e164PhoneNumber] = {
+    ...recipient,
+    kind: RecipientKind.Contact,
+    contactId: recipient.contactId ? recipient.contactId : uuidv1(),
+  }
+  addressToE164Number[data.address] = data.e164PhoneNumber
+
+  yield all([
+    put(setRecipientCache(recipientCache)),
+    put(storeLatestInRecents(recipient)),
+    put(updateE164PhoneNumberAddresses({}, addressToE164Number)),
```

Comment:
Why are you updating the mapping here?

Reply:
I updated the mapping here because `setRecipientCache` was already there. Would you suggest to move all those three to a more happy place?